src/components/hooks.js

```diff
@@ -130,60 +132,26 @@ export function useDeckCanvasSize() {
  * This hook handles a boolean isReady value,
  * which only returns true once every item in the
  * input list has been marked as "ready".
- * @param {string[]} items The items to wait on.
- * Should be defined as a constant
- * (outside a function component / render function),
- * otherwise strange bugs may occur.
- * @returns {array} An array
- * [isReady, setItemIsReady, setItemIsNotReady, resetReadyItems]
- * where isReady is the boolean value,
- * setItemIsReady marks one item as ready,
- * setItemIsNotReady marks one item as not ready,
- * and resetReadyItem marks all items as waiting.
+ * @param {string[]} statusValues The items to wait on.
+ * @returns {boolean} Whether the status values are all success.
  */
-export function useReady(supportedItems) {
-  const items = supportedItems;
-  const [waiting, setWaiting] = useState(items);
-
-  const setItemIsReady = useCallback((readyItem) => {
-    setWaiting((waitingItems) => {
-      const nextWaitingItems = waitingItems.filter(item => item !== readyItem);
-      // eslint-disable-next-line no-console
-      console.log(`cleared ${readyItem}; waiting on ${nextWaitingItems.length}: ${JSON.stringify(nextWaitingItems)}`);
-      return nextWaitingItems;
-    });
-  }, [setWaiting]);
-
-  const setItemIsNotReady = useCallback((notReadyItem) => {
-    setWaiting((waitingItems) => {
-      const nextWaitingItems = [...waitingItems, notReadyItem];
-      // eslint-disable-next-line no-console
-      console.log(`waiting on ${nextWaitingItems.length}: ${JSON.stringify(nextWaitingItems)}`);
-      return nextWaitingItems;
-    });
-  }, [setWaiting]);
-
-  const resetReadyItems = useCallback(() => {
-    setWaiting(items);
-    // eslint-disable-next-line no-console
-    console.log(`waiting on ${items.length}: ${JSON.stringify(items)}`);
-  }, [setWaiting, items]);
-
-  const isReady = waiting.length === 0;
-
-  return [isReady, setItemIsReady, setItemIsNotReady, resetReadyItems];
+export function useReady(statusValues) {
+  return useMemo(() => every(
+    statusValues, val => val === 'success',
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  ), statusValues);
 }
 
 /**
  * This hook manages a list of URLs,
  * with adding and resetting helpers.
  * @returns {array} An array
- * [urls, addUrl, resetUrls]
+ * [urls, addUrl]
  * where urls is the array of URL objects,
  * addUrl is a function for adding a URL to the array,
  * resetUrls is a function that clears the array.
  */
-export function useUrls() {
+export function useUrls(loaders, dataset) {
```

Comment:
What do `loaders` and `dataset` do here? They're needed to ensure that if they change, the urls go back to an empty array?  Should be in docs.